tools/misc/update_model_index.py

```diff
@@ -181,38 +214,47 @@ def parse_md(md_file):
 
     result = {'Collections': [collection], 'Models': models}
     yml_file = md_file[:-2] + 'yml'
-    with open(yml_file, 'w') as f:
-        mmcv.dump(result, f, file_format='yaml', sort_keys=False)
+
+    is_different = dump_yaml_and_check_difference(result, yml_file)
+    return is_different
 
 
 def update_model_index():
-    repo_root = osp.dirname(osp.dirname(__file__))
-    configs_dir = osp.join(repo_root, 'configs')
+    """Update model-index.yml according to model .md files.
+
+    Returns:
+        Bool: If the updated model-index.yml is different from the original.
+    """
+    configs_dir = osp.join(MMPOSE_ROOT, 'configs')
     yml_files = []
     for root, dirs, files in os.walk(configs_dir):
         if len(dirs) == 0:
             for file in files:
                 if file[-3:] == 'yml':
                     yml_files.append(os.path.join(root, file))
```

Comment:
Can this discovery logic replaced by glob? like

![image](https://user-images.githubusercontent.com/9464825/126578292-73fd51b7-4c4f-48f0-83a6-6c91f27c1b7a.png)

Reply:
Replaced by glob and tested.